src/parse.c

```diff
@@ -722,61 +722,123 @@ const char *qd_parse_annotations_v1(
         return parse_error;
     }
 
+    // define a shorthand name for the qd message annotation key prefix length
+#define QMPL QD_MA_PREFIX_LEN
+
+#define MIN(a,b) (((a)<(b))?(a):(b))
```

Comment:
ooh!  Please put this (and maybe MAX) into ctools.h - don't know how many times I've ended up doing this exact same thing.